docs/source/user_guide/writing-extractors.rst

```diff
@@ -0,0 +1,155 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_writing-extractors:
+
+Writing custom extractors
+*************************
+
+An extractor, in MetaLad sense, is a class derived from one of the base extractor classes defined in ``datalad_metalad.extractors.base`` (``DatasetMetadataExtractor`` or ``FileMetadataExtractor``).
+It needs to implement several required methods, most notably ``extract()``.
+
+Example extractors can be found in MetaLad source code.
+
+.. TODO:: Add links to the example extractors
+
+Base class
+==========
+
+There are two primary types of extractors, dataset-level and file-level.
+
+Dataset-level extractors, by inheritance from the ``DatasetMetadataExtractor`` class, can access the dataset on which they operate as ``self.dataset``. Extractor functions may use it to call any dataset methods.
+
+File-level extractors, by inheritance from the ``FileMetadataExtractor``, can refer to the file on which it operates through ``self.file_info``.
```

Comment:
In the section above, I wanted to clearly state the primary ways to query the dataset content. While dataset methods are fairly clear to me, I have pretty much no experience on `file_info` - maybe someone could add some details on what it can do (or a link to api docs).

Reply:
Added a little bit in 01622f8.